web/src/features/media/validation.ts

```diff
@@ -0,0 +1,92 @@
+import { z } from "zod";
+
+export enum MediaEnabledFields {
+  Input = "input",
+  Output = "output",
+  Metadata = "metadata",
+}
+
+export enum MediaContentType {
+  PNG = "image/png",
+  JPEG = "image/jpeg",
+  JPG = "image/jpg",
+  WEBP = "image/webp",
+  MP3 = "audio/mpeg",
+  MP3_LEGACY = "audio/mp3",
+  WAV = "audio/wav",
+  TXT = "text/plain",
+  PDF = "application/pdf",
+}
+
+export enum MediaFileExtension {
+  PNG = "png",
+  JPG = "jpg",
+  JPEG = "jpeg",
+  WEBP = "webp",
+  MP3 = "mp3",
+  MP4 = "mp4",
+  WAV = "wav",
+  TXT = "txt",
+  PDF = "pdf",
+}
+
+export const GetMediaUploadUrlQuerySchema = z.object({
+  traceId: z.string(),
+  observationId: z.string().nullish(),
+  contentType: z.nativeEnum(MediaContentType, {
+    message: `Invalid content type. Only supporting ${Object.values(
+      MediaContentType,
+    ).join(", ")}`,
+  }),
+  contentLength: z.number().positive().int(),
+  sha256Hash: z
+    .string()
+    .regex(
+      /^[A-Za-z0-9+/=]{44}$/,
+      "Must be a 44 character base64 encoded SHA-256 hash",
+    ),
+  field: z.nativeEnum(MediaEnabledFields, {
+    message: `Invalid field. Only supporting ${Object.values(
+      MediaEnabledFields,
+    ).join(", ")}`,
+  }),
+});
+
+export type GetMediaUploadUrlQuery = z.infer<
+  typeof GetMediaUploadUrlQuerySchema
+>;
+
+export const GetMediaUploadUrlResponseSchema = z.object({
+  uploadUrl: z.string().nullish(),
+  mediaId: z.string(),
+});
```

Comment:
do we need these responses necessarily in the SDKs? I would try to return as little as possible to SDKs to not have an API dependency here.

Reply:
Yes, we do. UploadURL necessary to upload the file, mediaID necessary to create the reference string with which e.g. the base64 string is replaced